src/redprl/refiner_types.fun

```diff
@@ -1840,149 +1750,96 @@ struct
+        val _ = Assert.levelLeq (l0, l1)
       in
-        |>:? goal #> (H, trivial)
+        T.empty #> (H, trivial)
       end
 
     fun InternalizeEq _ jdg =
       let
         val _ = RedPrlLog.trace "InternalizedEquality.InternalizeEq"
-        val H >> AJ.EQ ((m, n), (ty, l, k)) = jdg
+        val H >> AJ.EQ ((m, n), (ty, l)) = jdg
 
         (* the realizer must be `AX` *)
-        val (goal, _) = makeTrue H (Syn.into (Syn.EQUALITY (ty, m, n)), l, K.top)
-        val goalKind = makeTypeUnlessSubUniv H (ty, l, k) (l, K.top)
+        val (goal, _) = makeTrue H (Syn.into (Syn.EQUALITY (ty, m, n)), l)
       in
-        |>: goal >:? goalKind #> (H, trivial)
+        |>: goal #> (H, trivial)
       end
 
     (* (= ty a b) => a synth ~~> ty *)
-    fun SynthFromTrueEq z _ jdg =
+    (* this is for non-deterministic search *)
+    fun NondetSynthFromTrueEq z _ jdg =
       let
-        val _ = RedPrlLog.trace "InternalizedEquality.SynthFromTrueEq"
-        val H >> AJ.SYNTH (tm, l, k) = jdg
-        val AJ.TRUE (equal, l', _) = Hyps.lookup H z
+        val _ = RedPrlLog.trace "InternalizedEquality.NondetSynthFromTrueEq"
+        val H >> AJ.SYNTH (tm, l) = jdg
+        val AJ.TRUE (equal, l') = Hyps.lookup H z
         val Syn.EQUALITY (ty, a, b) = Syn.out equal
         val _ = Assert.alphaEqEither ((a, b), tm)
-        val goalKind = makeTypeUnlessSubUniv H (ty, l, k) (l', K.top)
       in
-        |>:? goalKind #> (H, ty)
+        T.empty #> (H, ty)
       end
 
     fun RewriteTrue sel eqterm alpha jdg =
       let
         val _ = RedPrlLog.trace "InternalizedEquality.RewriteTrue"
         val H >> catjdg = jdg
 
-        val (currentTy, l, k) =
+        val (currentTy, l) =
           case Selector.lookup sel (H, catjdg) of
              AJ.TRUE params => params
            | jdg => E.raiseError @@ E.NOT_APPLICABLE (Fpp.text "rewrite tactic", AJ.pretty jdg)
 
         val truncatedH = Selector.truncateFrom sel H
 
-        val (goalTyOfEq, holeTyOfEq) = makeSynth (truncatedH) (eqterm, NONE, K.top)
+        val (goalTyOfEq, holeTyOfEq) = makeSynth truncatedH (eqterm, l)
         val (goalTy, holeTy) = makeMatch (O.EQUALITY, 0, holeTyOfEq, [])
         val (goalM, holeM) = makeMatch (O.EQUALITY, 1, holeTyOfEq, [])
         val (goalN, holeN) = makeMatch (O.EQUALITY, 2, holeTyOfEq, [])
 
         val x = alpha 0
-        val truncatedHx = truncatedH @> (x, AJ.TRUE (holeTy, NONE, K.top))
-        val (motiveGoal, motiveHole) = makeTerm (truncatedHx) O.EXP
-        val motiveWfGoal = makeType (truncatedHx) (motiveHole, l, k)
+        val truncatedHx = truncatedH @> (x, AJ.TRUE (holeTy, l))
+        val (motiveGoal, motiveHole) = makeTerm truncatedHx O.EXP
+        val motiveWfGoal = makeType truncatedHx (motiveHole, l, K.top)
 
         val motiven = substVar (holeN, x) motiveHole
         val motivem = substVar (holeM, x) motiveHole
 
         val (H', catjdg') = Selector.map sel (fn _ => motiven) (H, catjdg)
```

Comment:
Here is the problem in `num.prl`; this replaces every slot of an atomic judgment with `motiven`. I think the logic we want is to replace a `TRUE (a, l)` judgment with the judgment `TRUE (motiven, l)`.